src/proto/NEAR.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+
+package TW.NEAR.Proto;
+option java_package = "wallet.core.jni.proto";
+
+message PublicKey {
+    uint32 key_type = 1;
+    bytes data = 2;
+}
+
+message FunctionCallPermission {
+    bytes allowance = 1; // uint128
+    string receiver_id = 2;
+}
+
+message FullAccessPermission {
+}
+
+message AccessKey {
+    uint64 nonce = 1;
+    oneof permission {
+        FunctionCallPermission function_call = 2;
+        FullAccessPermission full_access = 3;
+    }
+}
+
+message CreateAccount {
+}
+
+message DeployContract {
+    bytes code = 1;
+}
+
+message FunctionCall {
+    bytes method_name = 1;
+    bytes args = 2;
+    uint64 gas = 3;
+    bytes deposit = 4; // uint128
+}
+
+message Transfer {
+    bytes deposit = 1; // uint128
+}
+
+message Stake {
+    bytes stake = 1; // uint128
+    string public_key = 2;
+}
+
+message AddKey {
+    PublicKey public_key = 1;
+    AccessKey access_key = 2;
+}
+
+message DeleteKey {
+    PublicKey public_key = 1;
+}
+
+message DeleteAccount {
+    string beneficiary_id = 1;
+}
+
+message Action {
+    oneof payload {
+        CreateAccount create_account = 1;
+        DeployContract deploy_contract = 2;
+        FunctionCall function_call = 3;
+        Transfer transfer = 4;
+        Stake stake = 5;
+        AddKey add_key = 6;
+        DeleteKey delete_key = 7;
+        DeleteAccount delete_account = 8;
+    }
+}
+
+// Input data necessary to create a signed order.
+message SigningInput {
+    string signer_id = 1;
+    PublicKey public_key = 2;
+    uint64 nonce = 3;
+    string receiver_id = 4;
+    bytes block_hash = 5;
+    repeated Action actions = 6;
```

Comment:
can we keep proto as flat as possible?

Reply:
1. Why? Intention here is to keep protos similar to our actual transaction structure to avoid confusion.
2. Note sure what we actually can flatten here. E.g. creating an account is multiple actions, see:
https://github.com/nearprotocol/NEPs/pull/8/files?short_path=15b6752#diff-15b6752ec7d78e7b85b8c7de4a19cbd4
Note dapps can create transactions with any combination of these actions.

Reply:
to reduce client / app side integration effort, ideally we like a simple signing input interface, you can do whatever you want inside the C++ implementing

Reply:
@hewigovens but how to handle generic transaction signing then? Like app sends transaction and wallet signs it? Like in that case all actions need to be displayed.